File: app/models/snippet.js
const SnippetType = {
  PERSONAL: 'personal',
  MASTER: 'master'
};

class Snippet {
  constructor({
    id = null,
    title = '',
    content = '',
    folder = '',
    type = SnippetType.PERSONAL,
    description = '',
    order = 0,
    department = null
  } = {}) {
    this.id = id || this.generateId(folder, title, content);
    this.title = title;
    this.content = content;
    this.folder = folder;
    this.type = type;
    this.description = description;
    this.order = order;
    this.department = department;
  }

  generateId(folder, title, content) {
    const base = `${folder}_${title}_${content.substring(0, 100)}`;
    let hash = 0;
    for (let i = 0; i < base.length; i++) {
      hash = ((hash << 5) - hash) + base.charCodeAt(i);
      hash = hash & hash;
    }
    return `snippet_${Math.abs(hash).toString(36)}`;
  }
}

class SnippetFolder {
  constructor({
    id = null,
    name = '',
    snippets = [],
    order = 0
  } = {}) {
    this.id = id || Date.now().toString();
    this.name = name;
    this.snippets = snippets;
    this.order = order;
  }
}

module.exports = { Snippet, SnippetFolder, SnippetType };

File: SnipeeMac/SnipeeMac/SnipeeMac/Models/Snippet.swift
//
//  Snippet.swift
//  SnipeeMac
//

import Foundation

enum SnippetType: String, Codable {
    case personal
    case master
}

struct Snippet: Codable, Identifiable {
    let id: String
    var title: String
    var content: String
    var folder: String
    var type: SnippetType
    var description: String?
    var order: Int
    var createdAt: String?
    var updatedAt: String?
    
    init(id: String = UUID().uuidString, title: String, content: String, folder: String, type: SnippetType = .personal, description: String? = nil, order: Int = 0, createdAt: String? = nil, updatedAt: String? = nil) {
        self.id = id
        self.title = title
        self.content = content
        self.folder = folder
        self.type = type
        self.description = description
        self.order = order
        self.createdAt = createdAt ?? ISO8601DateFormatter().string(from: Date())
        self.updatedAt = updatedAt ?? ISO8601DateFormatter().string(from: Date())
    }
}

struct SnippetFolder: Codable, Identifiable {
    let id: String
    var name: String
    var snippets: [Snippet]
    var order: Int
    var updatedAt: String?
    
    init(id: String = UUID().uuidString, name: String, snippets: [Snippet] = [], order: Int = 0, updatedAt: String? = nil) {
        self.id = id
        self.name = name
        self.snippets = snippets
        self.order = order
        self.updatedAt = updatedAt ?? ISO8601DateFormatter().string(from: Date())
    }
}
